### PR TITLE
Add privilege management to bhyve

### PR DESCRIPTION
**Pull request by citrus-it**
## mail_msg

```

==== Nightly distributed build started:   Tue May 11 14:15:16 UTC 2021 ====
==== Nightly distributed build completed: Tue May 11 15:33:40 UTC 2021 ====

==== Total build time ====

real    1:18:23

==== Build environment ====

/usr/bin/uname
SunOS bloody 5.11 omnios-master-0721c7dd962 i86pc i386 i86pc

/opt/onbld/bin/i386/dmake
dmake: illumos make
number of concurrent jobs = 8

cw version 5.0
primary: /opt/gcc-7/bin/gcc
gcc (OmniOS 151039/7.5.0-il-1) 7.5.0
Copyright (C) 2017 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

shadow: /opt/gcc-4.4.4/bin/gcc
gcc (GCC) 4.4.4
Copyright (C) 2010 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

shadow: /data/omnios-build/omniosorg/bloody/illumos/usr/src/tools/proto/root_i386-nd/opt/onbld/bin/i386/smatch
0.6.1-rc1-il-5

/usr/jdk/openjdk11.0/bin/javac
openjdk full version "11.0.11+9-omnios-151039"

/usr/bin/openssl
OpenSSL 1.1.1k  25 Mar 2021
    API_COMPAT=0x10000000L

/usr/bin/as
as: Sun Compiler Common 12 SunOS_i386 snv_121 08/03/2009

/usr/ccs/bin/ld
ld: Software Generation Utilities - Solaris Link Editors: 5.11-1.1767 (illumos)

Build project:  default
Build taskid:   262

==== Nightly argument issues ====


==== Build version ====

omnios-bhyve-5b54da68f4e

==== Make clobber ERRORS ====


==== Make tools clobber ERRORS ====


==== Bootstrap build errors ====


==== Tools build errors ====


==== Build errors (non-DEBUG) ====


==== Build warnings (non-DEBUG) ====


==== Elapsed build time (non-DEBUG) ====

real    29:03.3
user  4:03:59.7
sys   1:19:11.6

==== Build noise differences (non-DEBUG) ====


==== package build errors (non-DEBUG) ====


==== Build errors (DEBUG) ====


==== Build warnings (DEBUG) ====


==== Elapsed build time (DEBUG) ====

real    24:43.5
user  3:26:52.6
sys   1:10:41.5

==== Build noise differences (DEBUG) ====


==== package build errors (DEBUG) ====


==== Validating manifests against proto area ====


==== Check versioning and ABI information ====


==== Check ELF runtime attributes ====


==== Diff ELF runtime attributes (since last build) ====


==== cstyle/hdrchk errors ====


==== Find core files ====


==== Check lists of files ====


==== Impact on file permissions ====


==== Linting packages ====
```

### CODE REVIEW

**Review by hadfl — APPROVED** [review]

File: usr/src/cmd/bhyve/bhyverun.c [hunk 1]
```diff
@@ -110,6 +110,9 @@ __FBSDID("$FreeBSD$");
 #include "rtc.h"
 #include "vga.h"
 #include "vmgenc.h"
+#ifndef __FreeBSD__
```
Comment (oetiker) [inline usr/src/cmd/bhyve/bhyverun.c:113]: should't this reather be a positive condition :) or is the privileges.h inherently portable, just not to FreeBDS ?

File: usr/src/cmd/bhyve/Makefile [hunk 2]
```diff
@@ -64,6 +64,7 @@ SRCS =	acpi.c			\
 	pctestdev.c		\
 	pm.c			\
 	post.c			\
+	privileges.c		\
```
Comment (oetiker) [inline usr/src/cmd/bhyve/Makefile:67]: how about namespacing the filename since all the internal functions seem to be namespaced ...

**Review by oetiker — CHANGES_REQUESTED** [review]

File: usr/src/cmd/bhyve/bhyverun.c (hunk 1 above)
Comment (citrus-it) [reply usr/src/cmd/bhyve/bhyverun.c:113]: This is how all of the illumos divergence from the upstream freebsd bhyve is done in illumos-gate - bracketed with __FreeBSD__

File: usr/src/cmd/bhyve/Makefile (hunk 2 above)
Comment (citrus-it) [reply usr/src/cmd/bhyve/Makefile:67]: I'm planning to just leave it as `privileges.[ch]`, and if the same file appears in the upstream freebsd source then we'll have to rename it then.

**Review by oetiker — APPROVED** [review]

**Pull request merged**